### DISCUSSION TITLE
Use python-semantic-release

### DISCUSSION THREAD

**Discussion by MaxGhenis** [PolicyEngine/policyengine-us-data#78] [Ideas]
https://python-semantic-release.readthedocs.io/en/latest/index.html

Suggestion at the bottom of how to emulate some of the yaml-changelog functionality: https://claude.ai/share/7f7999a0-4220-489a-a414-9d7b7b0a51ab

**Comment by nikhilwoodruff** [comment]
Seems like a complex move and the yaml changelog currently works. Is there a good reason to spend time on this?

**Comment by MaxGhenis** [comment]
Converted to a discussion over whether migrating to an established approach is worthwhile.

**Comment by anth-volk** [comment]
A couple benefits I've found digging through the package:
* More established, as you've indicated
* Potentially a bit less fragile, especially in the long run?

And a couple drawbacks:
* We'd likely have to migrate to using `poetry` for packaging over `setuptools_whatever`, which probably isn't a big deal, but could require some adjustment
* `python-semantic-release` versions based on commit, not on PR/push, and does so by parsing each commit (described [here](https://python-semantic-release.readthedocs.io/en/latest/commit-parsing.html)). Thus, we'd probably have to enforce specific commit message structuring, which could add some friction to the process, even for established contributors.
